### PR TITLE
Add Randomizer (from: Night , Improvements: LezaiYa)

### PR DESCRIPTION
**Pull request by LezaiYa**

### CODE REVIEW

**Comment by LezaiYa** [comment]
Don't merge it yet. I haven't tested it yet.

File: Patches/PlayerControlPatch.cs [hunk 1]
```diff
@@ -1159,6 +1159,64 @@ public static bool RpcCheckAndMurder(PlayerControl killer, PlayerControl target,
                     return false;
                 }
                 break;
+            //击杀萧暮
+            case CustomRoles.XiaoMu:
+                var Fg = IRandom.Instance;
+                int XiaoMu = Fg.Next(1, 4);
```
Comment (Gurge44) [inline Patches/PlayerControlPatch.cs:1165]: IRandom.Instance.Next "Generates a random number greater than or equal to 0 and less than maxValue." - So in this case, 4 will never be picked. Make it 1 to 5.

Comment (LezaiYa) [reply Patches/PlayerControlPatch.cs]: fix

**Comment by LezaiYa** [comment]
All code is complete and can now be merged

File: Patches/PlayerControlPatch.cs [hunk 2]
```diff
@@ -1159,6 +1159,65 @@ public static bool RpcCheckAndMurder(PlayerControl killer, PlayerControl target,
...
+                        Main.AllPlayerSpeed[killer.PlayerId] = Main.MinSpeed;
+                        ReportDeadBodyPatch.CanReport[killer.PlayerId] = false;
+                        killer.MarkDirtySettings();
+                        _ = new LateTask(() =>
+                        {
+                            Main.AllPlayerSpeed[killer.PlayerId] = Main.AllPlayerSpeed[killer.PlayerId] - Main.MinSpeed + tmpSpeed1;
+                            ReportDeadBodyPatch.CanReport[killer.PlayerId] = true;
+                            killer.MarkDirtySettings();
+                            RPC.PlaySoundRPC(killer.PlayerId, Sounds.TaskComplete);
+                        }, Options.TrapperBlockMoveTime.GetFloat(), "Trapper BlockMove");
```
Comment (Tommy-XL) [inline Patches/PlayerControlPatch.cs:1196]: It looks like the freezing time depends directly on the Trapper settings, it is better to set the time regardless of the Trapper settings (like setting or specifying it manually)

**Comment by 0xDrMoe** [comment]
Please make the internal name the same as the en_US variant. I do not want to dig for names internally when coding

**Comment by LezaiYa** [comment]
The code has been modified, please review

**Comment by Tommy-XL** [comment]
As Moe wrote above, XiaoMu needs to be changed to Randomizer in code

**Comment by LezaiYa** [comment]
> As Moe wrote above, XiaoMu needs to be changed to Randomizer in code

Ok, I'm adding a new effect now

**Review by Tommy-XL — APPROVED** [review]

**Pull request merged**